### PR TITLE
Fix compiling tests with profiling enabled for HIP

### PR DESCRIPTION
**Pull request by masterleinad**
Without this, I get
```
clang-10: error: ROCm agent detector could not identify any valid targets; please specify the target explicitly by passing a valid value to -amdgpu-target
clang-10: error: -amdgpu-target argument 'gfx000' is not recognized; using gfx803 instead [-Werror,-Winvalid-command-line-argument]
```
We have the same workaround a few lines earlier in this file.

### CODE REVIEW

File: core/unit_test/CMakeLists.txt [hunk 1]
```diff
@@ -341,6 +341,9 @@ if(Kokkos_ENABLE_PROFILING)
     tools/printing-tool.cpp
   )
 
+  # FIXME_HIP
```
Comment (crtrott) [inline core/unit_test/CMakeLists.txt:344]: Wait no lets not do this. Can we please just use the Kokkos_ARCH_VEGA906 or whatever option and add that to the nightly test?

Comment (masterleinad) [reply core/unit_test/CMakeLists.txt]: I just went with `KOKKOS_ADD_LIBRARY` instead.

File: core/unit_test/CMakeLists.txt [hunk 2]
```diff
@@ -336,9 +336,9 @@ KOKKOS_ADD_ADVANCED_TEST( UnitTest_PushFinalizeHook_terminate
 )
 
 if(Kokkos_ENABLE_PROFILING)
-  add_library(
-    printer-tool MODULE
-    tools/printing-tool.cpp
+  KOKKOS_ADD_LIBRARY(
```
Comment (DavidPoliakoff) [inline core/unit_test/CMakeLists.txt:339]: @jjwilke : is this going to cause problems elsewhere? I seem to remember the MODULE formulation being preferred

Comment (masterleinad) [reply core/unit_test/CMakeLists.txt:339]: I changed this since `TRIBITS` doesn't seem to allow `MODULE`.

Comment (dalg24) [reply core/unit_test/CMakeLists.txt:339]: You should have commented :/

Comment (dalg24) [reply core/unit_test/CMakeLists.txt:339]: This change seems to be causing the library to be installed with Kokkos

**Pull request merged**